nucypher/blockchain/eth/policies.py

```diff
@@ -175,23 +158,27 @@ def make_arrangements(self, network_middleware: RestMiddleware,
         selected_addresses = set()
         try:  # Sample by reading from the Blockchain
             actual_sample_quantity = math.ceil(target_sample_quantity * ADDITIONAL_URSULAS)
-            sampled_addresses = self.alice.recruit(quantity=actual_sample_quantity)
+            duration = int(calculate_period_duration(expiration))
+            sampled_addresses = self.alice.recruit(quantity=actual_sample_quantity,
+                                                   duration=duration,
+                                                   )
         except MinerAgent.NotEnoughMiners:
             error = "Cannot create policy with {} arrangements."
             raise self.NotEnoughBlockchainUrsulas(error.format(self.n))
         else:
             selected_addresses.update(sampled_addresses)
 
-        found_ursulas, spare_addresses = self.__find_ursulas(sampled_addresses, target_sample_quantity)
+        found_ursulas = self.__find_ursulas(sampled_addresses, target_sample_quantity)
 
-        candidates = handpicked_ursulas + found_ursulas
+        candidates = handpicked_ursulas
+        candidates.update(found_ursulas)
```

Comment:
Here we combine handpicked with sampled Ursulas for consideration, where there is no guarantee that handpicked Ursulas will be included in the policy if some of them reject, or are not assigned kfrags when the candidate pool is not exhausted.  IMO, this is good behaviour, but perhaps we can further prioritize the handpicked ursulas consideration.

Reply:
:+1:

Reply:
:heavy_check_mark: